File: source/ts/src/utils/angle.ts
export function normalize360(degrees: number): number {
  let r = degrees % 360;
  if (r < 0) r += 360;
  if (r >= 360) r -= 360;
  return r === 0 ? 0 : r;
}

export function degToRad(degrees: number): number {
  return degrees * (Math.PI / 180);
}

export function radToDeg(radians: number): number {
  return radians * (180 / Math.PI);
}

File: source/go/internal/utils/angle.go
package utils

import "github.com/ishankgupta95/panchang/source/go/v5/internal/jsnum"

func Normalize360(degrees float64) float64 {
	r := jsnum.Mod(degrees, 360)
	if r < 0 {
		r += 360
	}
	if r >= 360 {
		r -= 360
	}
	if r == 0 {
		return 0
	}
	return r
}

func DegToRad(degrees float64) float64 { return degrees * (jsnum.PI / 180) }

func RadToDeg(radians float64) float64 { return radians * (180 / jsnum.PI) }
